cnd/src/swap_protocols/rfc003/create_swap.rs

```diff
@@ -20,136 +20,45 @@ use genawaiter::{
     sync::{Co, Gen},
     GeneratorState,
 };
+use std::sync::Arc;
 
 /// Returns a future that tracks the swap negotiated from the given request and
-/// accept response on alpha ledger.
+/// accept response on a ledger.
 ///
 /// The current implementation is naive in the sense that it does not take into
 /// account situations where it is clear that no more events will happen even
 /// though in theory, there could. For example:
-/// - alpha funded
-/// - alpha refunded
+/// - funded
+/// - refunded
 ///
 /// It is highly unlikely for Bob to fund the HTLC now, yet the current
 /// implementation is still waiting for that.
-pub async fn create_alpha_watcher<D, A, AI, BI>(
+pub async fn create_watcher<D, S, L, A, H, I, T>(
     dependencies: D,
-    accepted: AcceptedSwap<A::AL, A::BL, A::AA, A::BA, AI, BI>,
+    ledger_state: Arc<S>,
```

Comment:
So I guess the problem why you are passing this separately is because you wouldn't know which instance of `LedgerState` to dispatch to if you just implemented it on the `Facade`? Would it make sense to assume that every caller of this function knows about the `Facade` itself and not just about `D`?

Differently asked, is every caller of this function something a component of `cnd` or do we need to ever call this from within the `COMIT` core?

If it is the former (which is what I expect) then we can get rid of the `From<>` trait bounds with `Arc`s and just reach into the facade and get the correct field out: `facade.alpha_ledger_states`.

I can send a follow-up PR about this :)

Reply:
I'll leave this one to you. I wasn't sure about making this function aware of the `Facade`, but it seems fine from your argument.

Reply:
Never mind, I'm tackling this as part of handling one of your other comments: https://github.com/comit-network/comit-rs/pull/2266#discussion_r393368253.